Сдача проектной работы №1

nsuhover opened the pull request:
Сдаю проектную работу №1

Commit:
Remove classes

Commit:
Change project name

nsuhover committed:
Написал программу для деления стоимости счета поровну между персонами.
Создан класс Calculator, который посчитывает общую введенную сумму товаров, делит её на равные части по количеству персон, а также определяет написание слов рубль и копейка в зависимости от введенной стоимости.
Создан класс ItemsBasket, который включает все в себя всю корзину введенных товаров за время рабоыт программы.

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,49 +1,101 @@
+        while (personsQuantity < 0) {personsQuantity = inputPersonsQuantity();} // получем колчиество персон
+        String itemName = inputItemName(1); // получаем имя первого товара
+        while (itemCost < 0) {itemCost = inputItemCost(itemName);} // получаем стоимость товара
+        new ItemsBasket();
```
Если ты используешь класс как утилиту со статическими методами, то ее не нужно инстанциировать. Относится к `ItemsBasket` и `Calculator`

gusar-off commented on src/main/java/ItemsBasket.java:
```diff
@@ -0,0 +1,20 @@
+public class ItemsBasket {
+    static String itemsNames = "None";
+
+    ItemsBasket() {
```
Лучше убрать

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,49 +1,101 @@
+        int errorInt = -1;
+        System.out.println("Введите число персон, на которых разделить счет:");
+        try {
+            int personsQuantity = scanner.nextInt(); // После запуска программа должна спрашивать у пользователя, на скольких человек необходимо разделить счёт.
```
У `Scanner` есть удобный метод `hasNextInt`, можно сразу узнать без дополнительных проверок возможно ли интерпретировать строку как `int`
Тогда достаточно будет сделать бесконечный цикл while(true) выход из которого осуществим только после получения правильного количества

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,49 +1,101 @@
-        double result = calculator.divideSum();
-        Formatter formatter = new Formatter();
+    public static String inputItemName(int type) {
+        Scanner scanner = new Scanner(System.in);
```
Лучшее переиспользовать инстанс сканнера, а не создавать каждый раз новый.

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,49 +1,101 @@
+        float errorFloat = -1.10f;
+
+        try {
+            itemCost = scanner.nextFloat();
```
У `Scanner` есть удобный метод `hasNextFloat`, можно сразу узнать без дополнительных проверок возможно ли интерпретировать строку как `float` 
Точно также как и с int, достаточно будет бесконечного цикла с условием выхода - прочтение правильного числа

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -1,22 +1,68 @@
 
-    double divideSum() {
-        return totalPrice / friendsCount;
+    public static String findRightPennyWord (int penny){
```
На самом деле по заданию достаточно определять окончание только для рублей

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -1,22 +1,68 @@
-        System.out.println(item.name + " в корзине");
+    public static String findRightRoubleWord (int roubles){
+        String roubleName;
+        int lastDigit = -1;
```
Тут инициализация значением не обязательна

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -1,22 +1,68 @@
+            roubleName = "рубля";
+            return roubleName;
+        }
+        else if ((lastDigit >= 5 && lastDigit <= 9)||(lastDigit == 0)){
```
Можно else тут сделать, остальные варианты предусмотрели ранее

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -1,22 +1,68 @@
+        String roubleName;
+        int lastDigit = -1;
+        int lastTwoDigits = -11;
+        lastTwoDigits = roubles - (roubles / 100) * 100;
```
Можно брать остаток от деления на 100 в данном случае, нагляднее имхо

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -1,22 +1,68 @@
+            roubleName = "рублей";
+            return roubleName;
+        }
+        lastDigit = roubles - (roubles / 10) * 10;
```
А тут на 10

gusar-off commented:
Есть еще замечания по код стилям, в Android студии можно сделать авто форматирование нажав ПКМ по нужному классу и выбрав `Reformat code`, или нажать сочетание клавиш из соответствующего пункта в меню.
![image](https://user-images.githubusercontent.com/118199521/205509787-8633baa3-4caf-4397-8a5a-92ba52b89a76.png)
Пройдись по всем классам с автоформатированием

nsuhover commented:
Привет! Спасибо за фидбек.
Пройдусь

С уважением,
Николай Суховерхов


вс, 4 дек. 2022 г. в 21:55, gusar-off ***@***.***>:

> Есть еще замечания по код стилям, в Android студии можно сделать авто
> форматирование нажав ПКМ по нужному классу и выбрав Reformat code, или
> нажать сочетание клавиш из соответствующего пункта в меню.
> [image: image]
> <https://user-images.githubusercontent.com/118199521/205509787-8633baa3-4caf-4397-8a5a-92ba52b89a76.png>
> Пройдись по всем классам с автоформатированием
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nsuhover/Java-Module-Project/pull/1#issuecomment-1336490829>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4LSIMJLPRL57G4OEEJWEVTWLTSLNANCNFSM6AAAAAASS6CRSI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

nsuhover committed:
Release notes :)
- убраны лишние инициализации значениями;
- в классе Calculator поиск последней цифры и двух цифр изменены на остаток деления от 100 и 10 соответственно;
- сокращена логика определения окончания слова "рубль", убрана логика по определению копеек, т.к. не требовалось по ТЗ;
- убран стандартный пустой конструктор ItemsBasket;
- убрана инициализация статических методов из классов ItemsBasket и Calculator;
- для обработки ошибок ввода пользователя заменена логика с try catch на hasNextInt и hasNextFloat;
